olp-cpp-sdk-dataservice-read/src/generated/api/StreamApi.cpp

```diff
@@ -87,9 +87,10 @@ StreamApi::SubscribeApiResponse StreamApi::Subscribe(
 
   OLP_SDK_LOG_DEBUG_F(kLogTag, "subscribe, uri=%s, status=%d",
                       metadata_uri.c_str(), http_response.status);
-
-  // TODO: Set x_correlation_id to the value received in http_response.header
-  // when http_response.header will be implemented.
+  if (!HandleCorrelationId(http_response.headers, x_correlation_id)) {
+    OLP_SDK_LOG_WARNING(kLogTag,
```

Comment:
I dont think this is so tragically to be handles as warning.

Reply:
X-Correlation should be in a separate pull request.